click_params/base.py

```diff
@@ -139,3 +140,33 @@ def convert(self, value, param, ctx):
 
     def __repr__(self):
         return self.name.upper()
+
+
+class EnumParamType(CustomParamType):
```

Comment:
we probably want to implement methods `get_missing_message` and `shell_complete` from the `click.Choice` class